core/src/main/java/org/mskcc/cbio/portal/servlet/EnrichmentsJSON.java

```diff
@@ -86,164 +99,169 @@ protected void doPost(HttpServletRequest httpServletRequest,
                 copyNumType = "del";
                 profileId = profileId.substring(0, profileId.length() - 4);
             }
-
-            //Get genetic profile ID (int) & Type
-            GeneticProfile gp = DaoGeneticProfile.getGeneticProfileByStableId(profileId);
-            final int gpId = gp.getGeneticProfileId();
-            String gpStableId = gp.getStableId();
-            String profileType = gp.getGeneticAlterationType().toString();
             
-            //Get cancer study internal id (int)
-            CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(cancerStudyId);
-            int cancerStudyInternalId = cancerStudy.getInternalId();
+            if(cancerStudyId != null) {
```

Comment:
Wow, this is some routine.   I'm wondering, given the size of the routine and all the nested constructs if its safer to check for authorization up at the top and punt there if the user has no access rather than try and wrap all the code here.